stdlib/public/core/CString.swift

```diff
@@ -150,6 +183,7 @@ extension String {
   ///     // Prints "nil"
   ///
   /// - Parameter cString: A pointer to a null-terminated UTF-8 code sequence.
+  @available(*, deprecated, renamed: "String.init(validatingCString:)")
```

Comment:
I settled on `@available(swift, deprecated: 6, ...)`, in that it will still be documented as deprecated, but the warning won't kick in until the language mode changes.